app/utils/validators/forms/FieldValidator.scala

```diff
@@ -116,11 +116,18 @@ object FieldValidator {
     (decimalPlaces: Int) =>
       (input: String) =>
         input.split('.') match {
-          case Array(a, b) if isNumeric(a) && isNumeric(b) => b.length <= decimalPlaces && (a + b).length <= totalLength && b.toDouble > 0
-          case Array(a) if isNumeric(a) && nonEmpty(a)     => a.length <= totalLength && a.toDouble > 0
+          case Array(a, b) if isNumeric(a) && isNumeric(b) => b.length <= decimalPlaces && (a + b).length <= totalLength
+          case Array(a) if isNumeric(a)                    => a.length <= totalLength
           case _                                           => false
   }
 
+  val validateDecimalGreaterThanZero: Int => Int => String => Boolean = (totalLength: Int) =>
+    (decimalPlaces: Int) =>
+      (input: String) =>
+        try { BigDecimal(input.trim()) > 0 && BigDecimal(input.trim()).scale < decimalPlaces && input.trim().replace(".", "").length < totalLength } catch {
+          case e: java.lang.NumberFormatException => false
```

Comment:
Still IMO we shouldn't trim value. Please 
```
try { 
       BigDecimal(input) > 0 && 
       BigDecimal(input).scale <= decimalPlaces && 
       input.length <= totalLength 
} catch {
          case e: java.lang.NumberFormatException => false
}
```

Reply:
I agree with what Patryk said. 
The `trim` was my suggestion. My fault guys! :)

Reply:
I would also format the code in the way Patryk did because it is easy for us to read the code and reason about it, than putting everything in one line.

Reply:
done